src/cytocherrypick/generate_test_data.py

```diff
@@ -36,7 +36,31 @@ def data_generation(output_file_path: str, num_of_rows: int = 100) -> pd.DataFra
             "image_name": ["image_" + str(i) for i in range(num_of_rows)],
         }
     )
+    return df
+
+
+# data writing
+def data_writing(output_file_path: str, df: pd.DataFrame) -> None:
+    """
+    This function writes a dataframe to a csv file.
+    The dataframe is generated by the data_generation function.
+
+    Parameters
+    ----------
+    output_file_path : str
+        Output file path for the test dataset csv file
+    df : pd.DataFrame
+        Mock DataFrame
+
+    Returns
+    -------
+    None
+    """
+    # Define paths
+    output_file_path = pathlib.Path(output_file_path)
+    # make directory if not exist
+    output_file_path.parent.mkdir(parents=True, exist_ok=True)
     # save the dataframe to a csv file
     df.to_csv(output_file_path, index=False)
 
-    return df
+    return None
```

Comment:
Consider removing this line. Python functions will automatically return `None` when there is no return specified.
```suggestion
```